edc-tests/e2e-tests/src/test/java/org/eclipse/tractusx/edc/lifecycle/DataWiper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.tractusx.edc.lifecycle;
+
+import org.eclipse.edc.connector.contract.spi.offer.store.ContractDefinitionStore;
+import org.eclipse.edc.connector.policy.spi.store.PolicyDefinitionStore;
+import org.eclipse.edc.spi.asset.AssetIndex;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to delete all objects from a runtime's data stores.
+ */
+public class DataWiper {
+
+    private final ServiceExtensionContext context;
+
+    public DataWiper(ServiceExtensionContext context) {
+        this.context = context;
+    }
+
+    public void clearPersistence() {
```

Comment:
I would suggest, as a long term goal, to avoid database cleanup on every test because it leads to biased behaviors, better to use random ids

Reply:
it will make assertions harder that are like `assertThat(resultList).hasSize(3)` or `...isEmpty()`, but yeah we could do it. we may even look at inject an instrumented store, which we can control during setup/teardown

Reply:
We'd likely have to filter the result to get the expected size. This would permit to parallelize tests, to make them running faster (before we'd need a comprehensive suite of tests though)

Reply:
Not sure if tests within the same class can be parallelized, but at least across test classes ('@Suite' maybe?), or possibly even modules

Reply:
Ideally the best case is to have them as parallelized as possible, anyway good for the moment, it could be something for the future.